sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -120,7 +120,13 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
       val plan = ShowTables(UnresolvedNamespace(ident), None)
       val ret = sparkSession.sessionState.executePlan(plan).toRdd.collect()
       val tables = ret
-        .map(row => ident ++ Seq(row.getString(1)))
+        .map(row =>
+          // for views, their namespace are empty
+          if (row.getString(0).isEmpty) {
+            Seq(row.getString(1))
+          } else {
+            ident ++ Seq(row.getString(1))
+          })
```

Comment:
For my understanding, is this a regression due to one of the recent commits like SPARK-39236?

Reply:
[SPARK-39236](https://issues.apache.org/jira/browse/SPARK-39236) updated `listTables(dbName)`. This PR does not cause regression on that JIRA. 

This is more like a side effect of [SPARK-39506](https://issues.apache.org/jira/browse/SPARK-39506). Because in [SPARK-39506](https://issues.apache.org/jira/browse/SPARK-39506) we support `setCurrentCatalog` and get `currentCatalog`, now for `listTables` it has a choice of which catalog to search for tables. In the past it always go to the only catalog which is `spark_catalog`, but now that catalog can be changed.

`listDatabases()` was already updated to respect the current catalog.

Reply:
Maybe it is hard to define whether this is a regression (I would rather it is a side effect given we introduce a way to control current catalog). I think at least it still maintains backwards compatibility. For old users who do not need set current catalog, it will still be the one that they would target to `spark_catalog`. The existing UT has tested that.

Reply:
Thanks for the detail. Yes, it's hard to say always during extending the existing semantics. New features are always nice to have, but what I hope is to keep the original features safe and independent as much as possible . As long as the old code works, we are good. Thank you again for all your efforts, @amaliujia .

Reply:
Avoiding deprecations is also the best way until we are sure that the new features are manure enough.

Reply:
agreed on that there should be a period of time to have new features mature enough with good adoptions before talking about deprecations.